src/RemoteRequest/CachedRemoteGetRequest.php

```diff
@@ -101,42 +102,48 @@ public function __construct(
 	 *
 	 * @param string $url URL to get.
 	 * @return Response Response for the executed request.
-	 * @throws FailedToGetFromRemoteUrl If retrieving the contents from the URL failed.
+	 * @throws FailedToGetFromRemoteUrl|FailedToGetCachedResponseData If retrieving the contents from the reponse failed.
```

Comment:
```suggestion
	 * @throws FailedToGetFromRemoteUrl|FailedToGetCachedResponseData If retrieving the contents from the response failed.
```

Reply:
Alright, I think what should happen here is that we provide an exception interface `FailedRemoteRequest` and have both `FailedToGetFromRemoteUrl` and `FailedToGetCachedResponseData` implement that.

That way, we can add a `@throws FailedRemoteRequest ...` to the `RemoteGetRequest` interface, and when we provide more specific exceptions in the implementations, we won't contradict the `RemoteGetRequest` interface.

Reply:
Done in https://github.com/ampproject/amp-wp/pull/4509/commits/af0b45013ee29a6704cf4b71504fde718ef7299d.